src/Microsoft.OData.Core/JsonLight/ODataJsonLightReaderNestedResourceInfo.cs

```diff
@@ -168,6 +174,33 @@ internal IEdmStructuredType NestedResourceType
             }
         }
 
+        /// <summary>
+        /// The resource type reference of the nested resource info which will be reported.
+        /// </summary>
+        internal IEdmStructuredTypeReference NestedResourceTypeReference
+        {
+            get
+            {
+                if (this.nestedResourceTypeReference == null)
+                {
+                    if (this.nestedResourceType == null && this.nestedProperty != null)
+                    {
+                        this.nestedResourceType = this.nestedProperty.Type.ToStructuredType();
+                    }
+
+                    bool resourceTypeNullable = true;
+                    if (nestedProperty != null && nestedProperty.Type != null)
+                    {
+                        resourceTypeNullable = nestedProperty.Type.IsNullable;
+                    }
+
+                    this.nestedResourceTypeReference = (nestedResourceType != null) ? nestedResourceType.ToTypeReference(resourceTypeNullable).AsStructured() : null;
+                }

```

Comment:
My suggestion:

1. Remove private fields:  "nestedResourceType" & "nestedResourceTypeReference".


2. Move these or similar codes into corresponding constructor, in first constructor, call " NestedResourceTypeReference = this.nestedProperty.Type.AsStructured(); (the name maybe different).

   in second construcotr, do same as above if the nested property not null, otherwise, 
NestedResourceTypeReference =nestedResourceType.ToTypeReference(true).AsStructured();


3. Add "internal IEdmStructuredTypeReference NestedResourceTypeReference { get; private set;}

4. Change "internal IEdmStrucutruedType NestedResourceType { get { return NestedResourceTypeReference.StructuredDefinition();}